postgresql/database.go

```diff
@@ -39,6 +40,7 @@ type Database interface {
 	builder.Builder
 
 	NewTransaction() (Tx, error)
+	With(interface{}) (Database, error)
```

Comment:
I think we should update interface declarations to include the arguments, it just makes it clearer, and go autocomplete daemon will show them. its especially helpful since we have so many interface{} argument types.

ie... change this line to... `With(sess interface{}) (Database, error)` and update everything else too